src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -62,85 +119,108 @@ int FilterParserThread::parseDATFilterFile()
         return ruleCount;
     }
 
-    unsigned int nbLine = 0;
-    while (!file.atEnd() && !m_abort) {
-        ++nbLine;
-        QByteArray line = file.readLine();
-        // Ignoring empty lines
-        line = line.trimmed();
-        if (line.isEmpty()) continue;
-        // Ignoring commented lines
-        if (line.startsWith('#') || line.startsWith("//")) continue;
+    static const int bufferSize = 2 * 1024 * 1024; // 2 MiB
+    char *buffer = new char[bufferSize];
+    qint64 bytesRead = 0;
+    int offset = 0;
+    int start = 0;
+    int endOfLine = -1;
+    int nbLine = 0;
+
+    while (true) {
+        memset(buffer + offset, 0, bufferSize - offset);
+        bytesRead = file.read(buffer + offset, bufferSize - offset);
+        if (bytesRead < 1) break;
```

Comment:
I think the following may happen: consider a file where the last line is not terminated with '\n'. Then this line will go into this spillover, `QFile::read()` returns 0 and we ignore it. That is why I prefer to replace newline characters with '\0' as well. [Here](https://gist.github.com/evsh/5b3b6f65c8127e759cd6db329a4ab214) is an example. Extracted from a component that reads and parses ASCII tables of much larger size, however.